execution_engine/src/core/engine_state/mod.rs

```diff
@@ -1635,17 +1637,26 @@ where
             .map_err(Error::from)
     }
 
-    pub fn put_trie(
+    pub fn put_trie_and_find_missing_descendant_trie_keys(
```

Comment:
I wonder whether `put_trie_and_find_missing_descendant_keys` or `put_trie_and_find_missing_descendants` would be any less clear.